### PR TITLE
feat: sync `CommunitySettings` across paired devices

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
This commit introduces a new `synced_at` field in the
`communities_settings` table so that it can be leveraged for syncing
community settings across devices.

It also adds sending and handling of sync community settings where
appropriate.

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

**Comment by 0x-r4bbit** [comment]
@richard-ramos @Samyoul only the first commit needs review, the second one is build artifacts from `make generate`. I'll squash them into the first one, once reviewed.

File: protocol/protobuf/application_metadata_message.proto [hunk 1]
```diff
@@ -61,5 +61,6 @@ message ApplicationMetadataMessage {
     SYNC_WALLET_ACCOUNT = 45;
     ACCEPT_CONTACT_REQUEST = 46;
     RETRACT_CONTACT_REQUEST = 47;
+    SYNC_COMMUNITY_SETTINGS = 49;
```
Comment (0x-r4bbit) [inline protocol/protobuf/application_metadata_message.proto:64]: There's another PR pending that introduces the `48` field here, so I already went with `49`

**Comment by 0x-r4bbit** [comment]
@cammellos I've updated this PR according to our discussion offline, namely:

1. Changed `synced_at` field to `clock` field in `CommunitySettings`
2. Extended `SynCommunity` to have a `settings` field which is of type `SyncCommunitySettings`, allowing us to send the settings along with a `SyncCommunity` message which reduces bandwidth
3. Notice that we still have an explicit sync message for communities settings which is used when editing a community, because in that case we're not sending sync messages for the community itself
3. Updated the tests accordingly, also added an explicit test for syncing community settings when editing communities as it needs special treatment

Still split up the changes in two commits, so reviewing the first one should be enough.

**Review by Samyoul — APPROVED** [review]
Nice PR, good tests

**Comment by 0x-r4bbit** [comment]
Looking into the failing test

**Comment by 0x-r4bbit** [comment]
Okay so it seems the issue is that there's no community in alice second device that has a synced_at value > 0, which means, alice second device might get the community description from the chat message (in which bob shares the community), but it has not received any trigger to get synced (which is required for it to have a sync_at >0)

Investigating further..

**Comment by 0x-r4bbit** [comment]
Alright, turned out that handling sync community settings was the culprit because we were sending malformed data.
Pushing an update

**Comment by 0x-r4bbit** [comment]
@churik Not sure if required, but pinging to this can be tested as per https://notes.status.im/pipeline-status-go

**Comment by qoqobolo** [comment]
Hi @PascalPrecht, e2e tests have caught the crash on the mobile app in this PR.
Take a look, please. 

### ISSUE 1: The app crashes when recovering multiaccount with a seed phrase

**Steps to reproduce:**
1. Install PR build from https://github.com/status-im/status-react/pull/13448#issuecomment-1145951415
2. Recover multiaccount with a seed phrase
3. Tap `Let's go`

**Logs:**
[crashlog_recoverseed.log](https://github.com/status-im/status-go/files/8833129/crashlog_recoverseed.log)


https://user-images.githubusercontent.com/67952253/171877366-870981b3-fb8e-4060-b0fd-137ddc22ef94.MP4

**Review by richard-ramos — APPROVED** [review]

**Comment by 0x-r4bbit** [comment]
Hey @qoqobolo thanks for letting me know!
I'll have a look at it today

**Comment by 0x-r4bbit** [comment]
@qoqobolo Hey :)

I could reproduce the bug and updated the PR accordingly.
Would you mind running your battery of tests again?

**Comment by qoqobolo** [comment]
> Would you mind running your battery of tests again?

Sure! I'll get back with the results.

**Comment by qoqobolo** [comment]
@PascalPrecht thanks for the fix! Works perfectly now.

**Comment by 0x-r4bbit** [comment]
@qoqobolo Thanks for confirming!

@Samyoul does this need a bump in `VERSION` before merge?

**Pull request merged**